test/integration/test_konnect_extension.go

```diff
@@ -0,0 +1,147 @@
+	}, testutils.ObjectUpdateTimeout, testutils.ObjectUpdateTick)
+
+	t.Logf("waiting for status.konnect and status.dataPlaneClientAuth to be set for KonnectExtension %s/%s", ke.Namespace, ke.Name)
+	require.EventuallyWithT(t, func(t *assert.CollectT) {
+		err := GetClients().MgrClient.Get(GetCtx(), types.NamespacedName{Name: ke.Name, Namespace: ke.Namespace}, ke)
+		require.NoError(t, err)
+		// Check Konnect control plane ID
+		assert.NotNil(t, ke.Status.Konnect, "status.konnect should be present")
+		assert.Equal(t, cp.GetKonnectID(), ke.Status.Konnect.ControlPlaneID, "Konnect control plane ID should be set in status")
+		// Check dataplane client auth
+		assert.NotNil(t, ke.Status.DataPlaneClientAuth, "status.dataPlaneClientAuth should be present")
+		assert.NotNil(t, ke.Status.DataPlaneClientAuth.CertificateSecretRef, "status.dataPlaneClientAuth.certiifcateSecretRef should be present")
+		assert.Equal(t, s.Name, ke.Status.DataPlaneClientAuth.CertificateSecretRef.Name,
+			"status.dataPlaneClientAuth.certiifcateSecretRef should have the expected secret name")
+
+	}, testutils.ObjectUpdateTimeout, testutils.ObjectUpdateTick)
+
+	// TODO: Create DataPlanes using the KonnectExtension after DP certifcates provisioning is done:
+	// https://github.com/Kong/gateway-operator/issues/874
+
+}
+
+func setKonnectExtensionDPCertSecretRef(t *testing.T, s *corev1.Secret) func(client.Object) {
+	return func(obj client.Object) {
+		ke, ok := obj.(*konnectv1alpha1.KonnectExtension)
+		require.True(t, ok)
+		ke.Spec.DataPlaneClientAuth = &konnectv1alpha1.DataPlaneClientAuth{
+			CertificateSecret: konnectv1alpha1.CertificateSecret{
+				Provisioning: lo.ToPtr(konnectv1alpha1.ManualSecretProvisioning),
+				CertificateSecretRef: &konnectv1alpha1.SecretRef{
+					Name: s.Name,
+				},
+			},
+		}
+	}
+}
+
+func checkKonnectExtensionConditions(t *assert.CollectT, ke *konnectv1alpha1.KonnectExtension) (bool, string) {
+	err := GetClients().MgrClient.Get(GetCtx(), types.NamespacedName{Name: ke.Name, Namespace: ke.Namespace}, ke)
+	require.NoError(t, err)
+
+	checkConditionTypes := []string{
+		konnectv1alpha1.ControlPlaneRefValidConditionType,
+		konnectv1alpha1.DataPlaneCertificateProvisionedConditionType,
+		konnectv1alpha1.KonnectExtensionReadyConditionType,
+	}
+	failedConditionTypes := make([]string, 0, len(checkConditionTypes))
+	conditionMap := lo.SliceToMap(ke.Status.Conditions, func(condition metav1.Condition) (string, metav1.ConditionStatus) {
+		return condition.Type, condition.Status
+	})
+
+	for _, conditionType := range checkConditionTypes {
+		status, ok := conditionMap[conditionType]
+		if !ok || status != metav1.ConditionTrue {
+			failedConditionTypes = append(failedConditionTypes, conditionType)
+		}
+	}
+
+	if len(failedConditionTypes) > 0 {
+		return false, fmt.Sprintf("condition(s) %s not set to True", strings.Join(failedConditionTypes, ", "))
+	}
+	return true, ""
+}
```

Comment:
Can we make this a generic (not necessarily using Go generics) function which checks a provided slice of conditions types for being set to true? Something that could be placed in https://github.com/Kong/gateway-operator/blob/27e877a3fa9f023b2bed8ad0c6a1fb289bbb86bb/pkg/utils/kubernetes/status.go or similar to be shared by other places that check conditions.

WDYT?